plugins/iommu/fu-iommu-device.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-iommu-device.h"
+
+struct _FuIommuDevice {
+	FuUdevDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuIommuDevice, fu_iommu_device, FU_TYPE_UDEV_DEVICE)
+
+static gboolean
+fu_iommu_device_probe (FuUdevDevice *device, GError **error)
+{
+	return fu_udev_device_set_physical_id (device, "iommu", error);
+}
+
+static gboolean
+fu_iommu_device_setup (FuDevice *device, GError **error)
+{
+	const gchar *path = fu_udev_device_get_sysfs_path (FU_UDEV_DEVICE (device));
+	g_autofree gchar *basename = g_path_get_basename (path);
+	g_autofree gchar *instance = g_strdup_printf ("system-iommu-%s", basename);
+	fu_device_add_instance_id (FU_DEVICE (device), instance);
```

Comment:
Do we need an actual device in the daemon? Can't we do the same trick we did in the pci-bcr plugin with `fu_plugin_udev_device_added()`?

Reply:
I think we can do that.  I'll switch it.

Reply:
fixed